### PR TITLE
Action user.send_invitation_email implemented

### PR DESCRIPTION
**Pull request by r-peschke**
Implemented including email base 
Detailed description see [wiki Action user.send_invitation_email](https://github.com/OpenSlides/OpenSlides/wiki/user.send_invitation_emaill)

### CODE REVIEW

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (jsangmeister) [reply openslides_backend/action/mixins/send_email_mixin.py:24]: Also, you can use an `Enum` here and instead of manually creating the `LIST` attribute, you can check if the string is present in the `__members__` map (https://docs.python.org/3/library/enum.html#iteration)

File: tests/system/action/user/test_email_client_server.py
Comment (jsangmeister) [inline tests/system/action/user/test_email_client_server.py:444]: remove before merge

File: tests/system/action/user/test_email_client_server.py
Comment (jsangmeister) [inline tests/system/action/user/test_email_client_server.py:436]:
```suggestion
                EmailMixin.check_email(email), "Email-address {} recognized as True"
```

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (r-peschke) [reply openslides_backend/action/mixins/send_email_mixin.py:21]: Done

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (r-peschke) [reply openslides_backend/action/mixins/send_email_mixin.py:24]: > Also, you can use an `Enum` here and instead of manually creating the `LIST` attribute, you can check if the string is present in the `__members__` map (https://docs.python.org/3/library/enum.html#iteration)
IMO an `Enum` isn't a really good alternative, because you have to decide whether you want `name` or `value` and have to write more;-).
The list is build also from class-attributes

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (r-peschke) [reply openslides_backend/action/mixins/send_email_mixin.py:45]: done

File: tests/system/action/user/test_email_client_server.py
Comment (r-peschke) [reply tests/system/action/user/test_email_client_server.py:444]: done

File: tests/system/action/user/test_email_client_server.py
Comment (r-peschke) [reply tests/system/action/user/test_email_client_server.py:436]: done

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (jsangmeister) [reply openslides_backend/action/mixins/send_email_mixin.py]: You can do `class ConnectionSecurity(str, Enum)` and then use it just like a string, so you don't have to write name/value. My main argument is that you double the code/values, and although these values will probably never change (or at least not very often), code duplication is always bad.

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (FinnStutzenstein) [inline openslides_backend/action/mixins/send_email_mixin.py:20]: Please give it a proper name

File: tests/system/action/user/test_email_client_server.py
Comment (FinnStutzenstein) [inline tests/system/action/user/test_email_client_server.py:1]: The location of this file does not fit; it is something more general since it only tests the email mixin.

File: tests/system/action/user/test_email_client_server.py
Comment (FinnStutzenstein) [inline tests/system/action/user/test_email_client_server.py:127]: Please transform this to a function. The class itself is not needed.

File: tests/system/action/user/test_email_client_server.py
Comment (FinnStutzenstein) [inline tests/system/action/user/test_email_client_server.py:155]: The users are not needed since you are only testing the email mixin and never the action

**Review by FinnStutzenstein — CHANGES_REQUESTED** [review]
Looks good so far, just some small remarks.

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (r-peschke) [reply openslides_backend/action/mixins/send_email_mixin.py:20]: done

File: tests/system/action/user/test_email_client_server.py
Comment (r-peschke) [reply tests/system/action/user/test_email_client_server.py:1]: moved to tests/system/action and renamed to test_send_email_mixin.py

File: tests/system/action/user/test_email_client_server.py
Comment (r-peschke) [reply tests/system/action/user/test_email_client_server.py:127]: done

File: tests/system/action/user/test_email_client_server.py
Comment (r-peschke) [reply tests/system/action/user/test_email_client_server.py:155]: Done

**Review by FinnStutzenstein — APPROVED** [review]

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (jsangmeister) [inline openslides_backend/action/mixins/send_email_mixin.py:64]:
```suggestion
        email_regex = r"[A-Z0-9._+\-ÄÖÜ]+@[A-Z0-9.\-ÄÖÜ]+\.[A-ZÄÖÜ]{2,}"
```

File: tests/system/action/user/test_send_invitation_email.py
Comment (jsangmeister) [inline tests/system/action/user/test_send_invitation_email.py:46]: It seems counterintuitive to me to execute the request in a special context for testing purposes, since this does not exist in case of "real" requests. Or how is it handled there?

**Comment by jsangmeister** [comment]
Can be merged once  you resolve the comments above, since they are either small details or questions

File: openslides_backend/action/actions/user/send_invitation_email.py
Comment (r-peschke) [reply openslides_backend/action/actions/user/send_invitation_email.py:38]: Done

File: openslides_backend/action/mixins/send_email_mixin.py
Comment (r-peschke) [reply openslides_backend/action/mixins/send_email_mixin.py:64]: Done

File: tests/system/action/user/test_send_invitation_email.py
Comment (r-peschke) [reply tests/system/action/user/test_send_invitation_email.py:46]: Maybe it's more the name, than the functionality: The "real" request uses a running mail server instance and doesn't need to start and stop it. In case of the test this context manager starts the aiosmtpd test server in a test specific way, i.e. testserver with or without auth (depends the EmailSettings) and especially stops it, depending on it's state.
I'll change the name from AiosmtpdConnectionManager to AiosmtpdServerManager

File: openslides_backend/action/actions/user/helper.py
Comment (r-peschke) [reply openslides_backend/action/actions/user/helper.py:18]: Yes, your  assumption was right, this was copied from OS3.
Changed with approval from emanuel

File: openslides_backend/action/actions/user/send_invitation_email.py
Comment (r-peschke) [reply openslides_backend/action/actions/user/send_invitation_email.py:198]: yes, it is better and more expressive.
Also see the description of email_mixin's send_email-method

File: openslides_backend/action/actions/user/send_invitation_email.py
Comment (r-peschke) [reply openslides_backend/action/actions/user/send_invitation_email.py:216]: done

File: tests/system/action/user/test_send_invitation_email.py
Comment (jsangmeister) [reply tests/system/action/user/test_send_invitation_email.py]: Ah okay, I just did not make the connection. I think both names are fine

**Pull request merged**
